claripy/ast/base.py

```diff
@@ -716,6 +719,9 @@ def replace_dict(self, replacements):
         #       raise ClaripyOperationError('cannot replace type %s ast with type %s ast' % (type(old), type(new)))
         #   old._check_replaceability(new)
 
+        if not self.variables:
```

Comment:
`set().issuperset(set())` returns True, so I think we *did* support replacing concrete ASTs.